src/engine/SCons/Tool/MSCommon/vc.py

```diff
@@ -161,11 +161,18 @@ def get_host_target(env):
     if not host_platform:
         host_platform = platform.machine()
 
+    # Solaris returns i86pc for both 32 and 64 bit architectures
```

Comment:
Why does this matter? Does it break some tests?

Reply:
All platforms are distinguishing between 32 and 64 bits, however, getting this info from `host_platform` and `_ARCH_TO_CANONICAL` is not possible on Solaris because `host_platform` itself is not enough to determine the architecture. That is why the canonical platform is being determined this way.